### ISSUE TITLE
Add FilterBar component

### ISSUE THREAD

**Issue by fbarl** [weaveworks/ui-components#192]
We currently use search bar with filters only in Workloads & Deploy status (2 different components):
![image](https://user-images.githubusercontent.com/1216874/39232850-e467e500-486e-11e8-850f-f06b588a9f39.png)

However, it might be good to take some ideas from Scope search bar and maybe make those 3 converge into one generalized component.
![image](https://user-images.githubusercontent.com/1216874/39234404-728da32a-4873-11e8-9157-14f5496bc12c.png)

In particular:
* Hitting _Enter_ on Deploy/Workloads search bar could wrap the query into a new closable bubble like Scope search currently does
* Adding _Filters_ dropdown to Scope search might take some of the load from the filters in bottom-left corner and at the same time move all filtering into one area of the screen (I think everything but namespace filters could be put in there). Then the default filters could already be shown in search bar like so:
![image](https://user-images.githubusercontent.com/1216874/39234654-2f28a8ea-4874-11e8-8561-5f721cf2e417.png)

cc @foot @bia

**Comment by bia** [comment]
+1 for consistency!

tl;dr : Deploy and Scope search should behave the same to look the same.

@foot and I just had a chat. The thing with the scope search style is that while you are typing it is filtering the whole view : 
![screen shot 2018-04-25 at 14 59 58](https://user-images.githubusercontent.com/92439/39247969-2d07c20e-489b-11e8-8a21-1e77211a67be.png)


Then when you commit to the search/bubble by pressing enter, all other screen elements disappear : 

![screen shot 2018-04-25 at 15 00 03](https://user-images.githubusercontent.com/92439/39248087-776139f2-489b-11e8-8d72-1ba94153f39c.png)


**Could we come up with a useful behaviour like that in Deploy?** Currently pressing enter after a query takes you to the page of the first workload (unless you've selected another result with the arrow keys) : 

![screen shot 2018-04-25 at 15 00 15](https://user-images.githubusercontent.com/92439/39248414-53205298-489c-11e8-9e91-ad1698633fbc.png)

Related, last year I had drawn some bubbled autofill suggestions for Monitor : 
![180425notebookfacetedsearch1](https://user-images.githubusercontent.com/92439/39249538-29ab0c5c-489f-11e8-803e-8602a5b93c5d.png)


Independent of that,
>Adding Filters dropdown to Scope search might take some of the load from the filters

would be an excellent idea.
